File: app/src/main/java/com/thesunnahrevival/sunnahassistant/views/SettingsActivity.java
package com.thesunnahrevival.sunnahassistant.views;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.ViewModelProviders;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.CompoundButton;

import com.thesunnahrevival.sunnahassistant.R;
import com.thesunnahrevival.sunnahassistant.utilities.NextReminderService;
import com.thesunnahrevival.sunnahassistant.viewmodels.SettingsViewModel;
import com.thesunnahrevival.sunnahassistant.databinding.ActivitySettingsBinding;

public class SettingsActivity extends AppCompatActivity implements View.OnClickListener, CompoundButton.OnCheckedChangeListener {

    private ActivitySettingsBinding mBinding;
    private SettingsViewModel mViewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_settings);
        mBinding.setLifecycleOwner(this);
        mViewModel = ViewModelProviders.of(this).get(SettingsViewModel.class);
        mBinding.setSettingsViewModel(mViewModel);
        if (getSupportActionBar() != null)
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSettings();
        mBinding.nexttReminderNotificationToggle.setOnCheckedChangeListener(this);
    }

    private void getSettings() {
        mViewModel.mSettings.observe(this, settings -> {
            if (settings != null) {
                mViewModel.setOffsetSavedValue(settings.getHijriOffSet());
                if (!settings.isShowNextReminderNotification())
                    mBinding.nexttReminderNotificationToggle.setChecked(false);
                if (!settings.isFirstLaunch())
                    mBinding.nexttReminderNotificationToggle.setVisibility(View.VISIBLE);
            }
        });
    }

    @Override
    public void onClick(View v) {
        if (v.getId() == R.id.change_hijri_date) {
            int offset = Integer.parseInt(mBinding.hijriOffsetValue.getText().toString().trim());
            mViewModel.updateHijriDate(offset);
        }
    }

    @Override
    public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
        if (mViewModel.mSettings.getValue() != null) {
            if (isChecked) {
                mViewModel.mSettings.getValue().setShowNextReminderNotification(true);
                startService(new Intent(this, NextReminderService.class));
            } else {
                mViewModel.mSettings.getValue().setShowNextReminderNotification(false);
                stopService(new Intent(this, NextReminderService.class));
            }
            mViewModel.updateSettings();
        }
    }
}

File: app/src/main/java/com/thesunnahrevival/sunnahassistant/views/SettingsActivity.kt
package com.thesunnahrevival.sunnahassistant.views

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import com.thesunnahrevival.sunnahassistant.R

class SettingsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)
        val transaction = supportFragmentManager.beginTransaction()
        val fragmentToShow = intent.getStringExtra(FRAGMENT_TO_SHOW) ?: "default"
        val fragment = if (fragmentToShow.matches(PRAYER_CONSTANT.toRegex()))
            SettingsFragment.newInstance(R.layout.prayer_time_settings)
        else if (fragmentToShow.matches(NOTIFICATION_SETTINGS.toRegex()))
            SettingsFragment.newInstance(R.layout.notification_settings)
        else
            SettingsFragment.newInstance(R.layout.settings_lists)
        transaction.replace(R.id.settings_container, fragment )
        transaction.commit()
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        when(item?.itemId){
            android.R.id.home -> super.onBackPressed()
        }
        return true
    }

    companion object {
        const val PRAYER_CONSTANT = "prayer"
        const val FRAGMENT_TO_SHOW = "fragmentToShow"
        const val NOTIFICATION_SETTINGS = "notifications"
    }

}